app/models/concerns/belongs_to_proxies.rb

```diff
@@ -17,7 +17,7 @@ def belongs_to_proxy(name, options)
     def register_smart_proxy(name, options)
       self.registered_smart_proxies = registered_smart_proxies.merge(name => options)
       belongs_to name, :class_name => 'SmartProxy'
-      validates name, :proxy_features => { :feature => options[:feature] }
+      validates name, :proxy_features => { :feature => options[:feature], :required => options[:required] }
```

Comment:
Isn't this `validates name, presence: true` reinvented?

Reply:
Ah rails and it's existential problems: presence does not validate existence.  If I create a realm with a proxy ID of 923 which doesn't exist, Foreman will happily create a Realm even though no such proxy even exists, nevermind has the Realm feature.

When a host goes to create a realm entry, orchestration will fail.

Reply:
Doesn't rails check for existence when an inverse_of is present? I'm not sure at all, just saying.

Reply:
I will check but I don't think so

Reply:
It doesn't seem to work that way, I think that's more about calling validates on the associated object, but there is no associated object in this case since the ID is bad.  It doesn't catch that.

I still get this

```
[root@centos7-devel ~]# hammer -u admin -p changeme realm create --name EXAMPLE.COM --realm-proxy-id 9452 --realm-type FreeIPA
Could not create the realm:
  PG::ForeignKeyViolation: ERROR:  insert or update on table "realms" violates foreign key constraint "realms_realm_proxy_id_fk"
  DETAIL:  Key (realm_proxy_id)=(9452) is not present in table "smart_proxies".
  : INSERT INTO "realms" ("name", "realm_type", "realm_proxy_id", "created_at", "updated_at") VALUES ($1, $2, $3, $4, $5) RETURNING "id"
```